### DISCUSSION TITLE
Strimzi 0.35.1 to 0.37.0 with KRaft upgrade strategy

### DISCUSSION THREAD

**Discussion by alexeytokar** [strimzi/strimzi-kafka-operator#9162] [Q&A] [answered]
Ok, so the current setup is:
* Strimzi 0.35.1 (supports Kafka version up to 3.4.0)
* Kafka 3.4.0 +KRaft

Desired state:
* Strimzi 0.37.0 (as it has scrum-sha auth support for KRaft)

The problem:
* I can't update Kafka to 3.5.0 as 0.35.1 doesn't support it
* I can't update to 0.37.0 first as KRaft can be used only with KafkaNodePools support
* I can't create KafkaNodePool as my cluster version 3.4.0 and it requires 3.5.0

What is my possible strategy here? What order will let me get me to the desired state 🤔

**Comment by ppatierno** [comment, marked as answer]
My first question is ...
You wrote you have `Kafka 3.4.0 +KRaft` ... what does it mean for you? For me it means you have a Kafka cluster on version 3.4.0 using KRaft which in Strimzi means your operator has both KafkaNodePools and UseKRaft feature gates enabled. Is it what you have?

As additional information, there is not yet support for Kafka upgrades when the cluster is running in KRaft mode. As well as, but it should not be the case, there is no support yet for migrating from ZooKeeper to KRaft.

The UseKRaft feature gate is just for testing and development purpose only, so upgrade and migration are not supported yet and it's going to change in the future mostly "breaking" already running KRaft clusters.

**Comment by alexeytokar** [comment]
> means your operator has both KafkaNodePools and UseKRaft feature gates enabled. Is it what you have?

unfortunately no as KafkaNodePools available since 0.36.0 and I run 0.35.1 operator :(


> so upgrade and migration are not supported yet

ok. I guess this is the answer :'( :)

**Comment by ppatierno** [comment]
> unfortunately no as KafkaNodePools available since 0.36.0 and I run 0.35.1 operator :(

So what did you mean by `Kafka 3.4.0 +KRaft` how was your Kafka 3.4.0 cluster running in KRaft mode with Strimzi 0.35.1? Maybe I just mislead what you meant by `Kafka 3.4.0 +KRaft` so I was curious asking.

**Reply by alexeytokar** [reply]
I'm running Strimzi operator 0.35.1 with Kafka version 3.4.0 and [STRIMZI_FEATURE_GATES="+UseKRaft"](https://strimzi.io/docs/operators/0.35.1/deploying#ref-operator-use-kraft-feature-gate-str)

It was deployed before 0.36.0 release and without support of KafkaNodePools

edited:
basically it says:

> Upgrades and downgrades of Apache Kafka versions or the Strimzi operator are not supported. Users might need to delete the cluster, upgrade the operator and deploy a new Kafka cluster.

so the answer "it is not supported" is acceptable as I was notified :) but it still looks like it can be overcome if migration to KafkaNodePools were done in a separate step of the Kafka version upgrade 🤔

**Reply by ppatierno** [reply]
Ah ok! So you were using the KRaft feature gate as it was working before without node pools.
Yeah I don't think there is a smooth migration path for what you are asking.
@scholzj anything I am missing here?

**Reply by scholzj** [reply]
I did not try anything about upgrades of existing KRaft clusters. So I do not know if there are some possibilities to work around it or not. I assume you can try to follow the migration guide for node pools ad see if it works or not. But the general expectation for Kraft right now is that you just delete it and deploy a new cluster.
